opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1380,7 +1380,7 @@ namespace Opm {
     int
     BlackoilWellModel<TypeTag>::numComponents() const
     {
-        if (numWells() > 0 && numPhases() < 3) {
```

Comment:
This is the very hacky code without comments, which makes it hard. 

Should we check whether we have wells here at all? 

I mean, why do not we just do `if (numPhases() < 3) {`, which is still hacky while simpler.

Reply:
I will do some tests.  The PR looks okay. I will just check to answer my own question.

Reply:
maybe the original author should comment on whether `numWells()` is needed and what for.

My impression is that this cannot depend on whether there are wells or not and the current version is still buggy if there are no wells at all. The return value of `numComponents` governs the size of `B_avg` used in [`computeAverageFormationFactor`](/OPM/opm-simulators/blob/f95718a3c5d8254053f4ee5097a340d1b8461fc8/opm/simulators/wells/BlackoilWellModel_impl.hpp#L1324-L1333) and that is done for all cells even if there are no wells anywhere. The call of these functions is in `computeWellPotentials` which is again called in any case (whether there are wells or not).